src/Raven.Client/Documents/Linq/QueryMethodConverter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Linq.Expressions;
+using Raven.Client.Documents.Session;
+
+namespace Raven.Client.Documents.Linq
+{
+    public abstract class QueryMethodConverter
+    {
+        public struct Parameters<T>
```

Comment:
Could be a `readonly struct` with readonly fields and then `TryConvertQueryMethod` could take as `in` parameter?

Reply:
Wouldn't the implication be that the person referencing the client API would need to switch the project to C# 7.2+ in order to implement the converter?